### PR TITLE
Quarkus version

### LINKED ISSUE #20: Quarkus version?
Would you be interested in having a Quarkus branch? I'd be open to working on it if so.

### PR DESCRIPTION
**Pull request by edeandrea**
This is a Quarkus version of the application. Currently I can't get it to run.

Also I don't want to merge this against `main`. I want to merge it against another branch (`quarkus` maybe?), but that branch doesn't yet exist.

It builds, but when I try and load the main page I'm seeing this error & I can't figure out how to fix it:

```
2024-10-31 08:59:57,395 ERROR [com.vaa.flo.ser.DefaultErrorHandler] (executor-thread-1) : java.lang.IllegalArgumentException: Unable to create an instance of 'com.vaadin.flow.server.auth.MenuAccessControl'. Make sure the class has a public no-arg constructor.
        at com.vaadin.flow.internal.ReflectTools.createProxyInstance(ReflectTools.java:520)
        at com.vaadin.flow.internal.ReflectTools.createInstance(ReflectTools.java:452)
        at com.vaadin.flow.di.DefaultInstantiator.create(DefaultInstantiator.java:198)
        at com.vaadin.flow.di.DefaultInstantiator.getOrCreate(DefaultInstantiator.java:72)
        at com.vaadin.quarkus.QuarkusInstantiator.lambda$getOrCreate$2(QuarkusInstantiator.java:77)
        at com.vaadin.quarkus.BeanLookup.lookupOrElseGet(BeanLookup.java:94)
        at com.vaadin.quarkus.QuarkusInstantiator.getOrCreate(QuarkusInstantiator.java:76)
        at com.vaadin.flow.di.Instantiator.getMenuAccessControl(Instantiator.java:217)
        at com.vaadin.hilla.route.RouteUnifyingIndexHtmlRequestListener.collectServerViews(RouteUnifyingIndexHtmlRequestListener.java:191)
        at com.vaadin.hilla.route.RouteUnifyingIndexHtmlRequestListener.modifyIndexHtmlResponse(RouteUnifyingIndexHtmlRequestListener.java:114)
        at com.vaadin.flow.server.VaadinService.lambda$modifyIndexHtmlResponse$5(VaadinService.java:619)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
        at com.vaadin.flow.server.VaadinService.modifyIndexHtmlResponse(VaadinService.java:618)
        at com.vaadin.flow.server.communication.IndexHtmlRequestHandler.synchronizedHandleRequest(IndexHtmlRequestHandler.java:200)
        at com.vaadin.flow.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40)
        at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1584)
        at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:398)
        at com.vaadin.quarkus.QuarkusVaadinServlet.service(QuarkusVaadinServlet.java:84)
        at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:614)
        at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
        at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:63)
        at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
        at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
        at io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:67)
        at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:133)
        at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
        at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
        at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:65)
        at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
        at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
        at io.undertow.security.handlers.NotificationReceiverHandler.handleRequest(NotificationReceiverHandler.java:50)
        at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
        at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:247)
        at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:111)
        at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:108)
        at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
        at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
        at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$8$1.call(UndertowDeploymentRecorder.java:643)
        at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:227)
        at io.undertow.servlet.handlers.ServletInitialHandler.handleRequest(ServletInitialHandler.java:152)
        at io.undertow.server.handlers.CanonicalPathHandler.handleRequest(CanonicalPathHandler.java:49)
        at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$1.handleRequest(UndertowDeploymentRecorder.java:126)
        at io.undertow.server.Connectors.executeRootHandler(Connectors.java:284)
        at io.undertow.server.DefaultExchangeHandler.handle(DefaultExchangeHandler.java:18)
        at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$4$2.run(UndertowDeploymentRecorder.java:443)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:627)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
        at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:1583)
```

Closes #20

### CODE REVIEW

**Comment by edeandrea** [comment]
I've gotten this to work. Again though I don't want to merge to `main`. If you create a `quarkus` branch in your repo I can then target the PR against it.

I've even added some tests using Playwright (see https://github.com/edeandrea/java-ai-playground/actions/runs/11622248355)

**Review by Re2906 — APPROVED** [review]

**Comment by marcushellberg** [comment]
Hey. Thanks for the PR! I've been on the road for two weeks and a bit behind on everything. I'll take a look next week.

File: pom.xml [hunk 1]
```diff
@@ -1,174 +1,241 @@
...
-        </dependencies>
-    </dependencyManagement>
+  <properties>
+    <assertj.version>3.26.3</assertj.version>
+    <awaitility.version>4.2.2</awaitility.version>
+    <compiler-plugin.version>3.13.0</compiler-plugin.version>
+    <maven.compiler.release>17</maven.compiler.release>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+    <quarkus.hilla.version>24.5.0-beta1</quarkus.hilla.version>
```
Comment (mcollovati) [inline pom.xml]: You can upgrade to 24.5.0 now

Comment (edeandrea) [reply pom.xml]: Awesome! I'm away from my computer for the rest of the day but will take care of that 1st thing in the morning!

**Review by Re2906 — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (edeandrea) [reply pom.xml]: Done!

**Comment by marcushellberg** [comment]
Seems to work. Created a `quarkus` branch and merged. Thanks again for your contribution! 🙏

**Comment by edeandrea** [comment]
You're welcome! If you don't mind, I'm going to now push a PR which adds in dependabot and a workflow which will run the tests so that versions of things on this branch stay up-to-date automatically.

**Pull request merged**